presentation/src/main/java/com/architect/coders/mu8/categories/CategoriesActivity.kt

```diff
@@ -34,22 +41,25 @@ class MainActivity : AppCompatActivity() {
 
         recycler = findViewById(R.id.marvel_list)
 
-        viewModel = ViewModelProviders.of(
-            this,
-            CategoriesViewModelFactory(CategoriesRepository())
-        )[CategoriesViewModel::class.java]
-
+        viewModel = getViewModel { CategoriesViewModel(CategoriesRepository()) }
         viewModel.model.observe(this, Observer(::updateUi))
     }
 
     private fun updateUi(model: UiModel) {
         when (model) {
             is UiModel.Content -> {
-                adapter = CategoriesAdapter(viewModel::onMoviewClicked, model.categories)
+                adapter = CategoriesAdapter(viewModel::onCategoryClicked, model.categories)
                 recycler.layoutManager = LinearLayoutManager(this)
```

Comment:
Assign it in the onCreate to avoid more that once.
recycler.layoutManager = LinearLayoutManager(this)
 

 
Assign it in the onCreate to avoid more than once.
```
recycler.adapter = adapter	                
recycler.adapter = adapter
```

Reply:
Good catch, I'll correct it.

C.C. @SalosV this was your code :P